numba_cuda/numba/cuda/lowering.py

```diff
@@ -120,3 +119,38 @@ def store_var_needed(self, name):
             # lowering with debuginfo
             or self._disable_sroa_like_opt
         )
+
+    def delvar(self, name):
+        """
+        Delete the given variable.
+        """
+        if name in self.poly_var_set:
+            if (
+                self._cur_ir_block == self.blocks[self.lastblk]
+                and not self.poly_cleaned
+            ):
+                # Decref, zero-fill the debug union for polymorphic only
+                # at the last block
+                for k, v in self.poly_var_loc_map.items():
+                    self.decref(self.typeof(k), self.builder.load(v))
+                    self.builder.store(Constant(v.type.pointee, None), v)
+                    self.poly_cleaned = True
+            return
+
+        super().delvar(name)
+
+    def getvar(self, name):
+        """
+        Get a pointer to the given variable's slot.
+        """
+        if name in self.poly_var_set:
+            index = name.find(".")
+            src_name = name[:index] if index > 0 else name
```

Comment:
A tiny nit, maybe a simpler way to do this in Python:

```suggestion
            src_name = name.split(".")[0]
```

Reply:
Resolved in https://github.com/NVIDIA/numba-cuda/pull/305/commits/ea455212d51eb8feb988e95a36bde1192869ea5d